### PR TITLE
III-4290 Refactor updating contact point of an organizer

### PR DESCRIPTION
**Pull request by LucWollants**
### Added
- Added new `UpdateContactPoint` command handler
- Added new `UpdateContactPoint` request handler

### Removed
- Removed obsolete code for handling UpdateContactPoint

---
Ticket: https://jira.uitdatabank.be/browse/III-4290

### CODE REVIEW

**Comment by LucWollants** [comment]
I did not refactor `Organizer::updateContactPoint` to use the `ContactPoint` class from the `Model` namespace as this would lead to a lot of conversions because that method is also used inside `Organizer::create`.

File: src/Http/Organizer/LegacyContactPointRequestBodyParser.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Organizer;
+
+use CultuurNet\UDB3\Http\Request\Body\RequestBodyParser;
+use Psr\Http\Message\ServerRequestInterface;
+
+class LegacyContactPointRequestBodyParser implements RequestBodyParser
```
Comment (bertramakers) [inline src/Http/Organizer/LegacyContactPointRequestBodyParser.php:10]: Nit: In other places this is usually called `BackwardCompatible...RequestBodyParser`. Either name is fine by me but maybe rename the others if you prefer `Legacy...RequestBodyParser`.

File: src/Organizer/Serializers/UpdateContactPointDenormalizer.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    public function __construct(string $organizerId)
+    {
+        $this->organizerId = $organizerId;
+    }
+
+    public function denormalize($data, $type, $format = null, array $context = []): UpdateContactPoint
+    {
+        $telephoneNumbers = [];
+        if (isset($data['phone'])) {
+            foreach ($data['phone'] as $phone) {
```
Comment (bertramakers) [inline src/Organizer/Serializers/UpdateContactPointDenormalizer.php:30]: Nit: I think you can remove the `if (isset(...))` by doing this:

```
foreach ($data['phone'] ?? [] as $phone) {
  // ...
}
```

Comment (bertramakers) [reply src/Organizer/Serializers/UpdateContactPointDenormalizer.php:30]: Or use `array_map` so you don't need the `$telephoneNumbers = [];` and make it a one-liner.

For example:

```
$telephoneNumbers = array_map(fn ($phone) => new TelephoneNumber($phone), $data['phone'] ?? []);
```

File: tests/Http/Organizer/UpdateContactPointRequestHandlerTest.php [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+                    new EmailAddresses(new EmailAddress('info@publiq.be')),
+                    new Urls(new Url('https://www.publiq.be'))
+                ),
+            ],
+            'update multiple contact point information' => [
+                '{
+                    "phone": ["016 10 20 30", "016 11 22 33"],
+                    "email": ["info@publiq.be", "info@cn.be"],
+                    "url": ["https://www.publiq.be", "https://www.cn.be"]
+                }',
```
Comment (bertramakers) [inline tests/Http/Organizer/UpdateContactPointRequestHandlerTest.php:117]: More a general comment / FYI: I'm definitely okay with hardcoding JSON in the tests, but if it becomes too complex to write you can also use `Json::encode([... your data as an assoc array ...])` if you want

Comment (bertramakers) [reply tests/Http/Organizer/UpdateContactPointRequestHandlerTest.php:117]: And/or add a `withBodyFromArray()` or something like that to `Psr7RequestBuilder` that just takes an assoc array and converts it to Json so we don't have to repeat that in every test file.

File: tests/Http/Organizer/UpdateContactPointRequestHandlerTest.php [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+            ->build('PUT');
+
+        $this->assertCallableThrowsApiProblem(
+            ApiProblem::bodyInvalidData(
+                new SchemaError('/phone/0', 'Minimum string length is 1, found 0')
+            ),
+            fn () => $this->updateContactPointRequestHandler->handle($updateUrlRequest)
+        );
+    }
+}
```
Comment (bertramakers) [inline tests/Http/Organizer/UpdateContactPointRequestHandlerTest.php:268]: Very extensive tests, nice 👍

**Review by bertramakers — APPROVED** [review]
Looks great, just some minor suggestions or general comments. Nothing blocking or that _needs_ to be picked up, so approving already

File: src/Http/Organizer/LegacyContactPointRequestBodyParser.php (hunk 1 above)
Comment (LucWollants) [reply src/Http/Organizer/LegacyContactPointRequestBodyParser.php:10]: Nicely spotted. This was indeed a sub comment inside this commit https://github.com/cultuurnet/udb3-silex/pull/791/commits/5bd28e03cc52a4beb5a470d5cc14325b31d708e9

Comment (bertramakers) [reply src/Http/Organizer/LegacyContactPointRequestBodyParser.php:10]: Ah I overlooked that because I reviewed the complete diff now for brevity's sake 😉

Comment (LucWollants) [reply src/Http/Organizer/LegacyContactPointRequestBodyParser.php:10]: But both names are fine for me. You can pick what you prefer 😄

File: src/Organizer/Serializers/UpdateContactPointDenormalizer.php (hunk 2 above)
Comment (LucWollants) [reply src/Organizer/Serializers/UpdateContactPointDenormalizer.php]: The `array_map` is indeed much more elegant.

File: tests/Http/Organizer/UpdateContactPointRequestHandlerTest.php (hunk 3 above)
Comment (LucWollants) [reply tests/Http/Organizer/UpdateContactPointRequestHandlerTest.php]: I added this new method `withBodyFromArray`.

File: src/Http/Organizer/LegacyContactPointRequestBodyParser.php (hunk 1 above)
Comment (LucWollants) [reply src/Http/Organizer/LegacyContactPointRequestBodyParser.php:10]: As discussed on Slack I took the `Legacy` variant

**Pull request merged**